File: packages/envsync-cli/internal/repository/responses/env_type.go
package responses

import "time"

// EnvTypeResponse represents the response structure for environment types
type EnvTypeResponse struct {
	ID          string    `json:"id"`
	OrgID       string    `json:"org_id"`
	Name        string    `json:"name"`
	AppID       string    `json:"app_id"`
	IsDefault   bool      `json:"is_default"`
	IsProtected bool      `json:"is_protected"`
	Color       string    `json:"color"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: packages/envsync-stdb/src/tables/env_type.rs
use spacetimedb::{table, Timestamp};

/// Environment type — replaces PostgreSQL `env_type` table.
#[table(public, accessor = env_type)]
pub struct EnvType {
    #[primary_key]
    #[auto_inc]
    pub id: u64,
    /// UUID string used as external identifier
    pub uuid: String,
    #[index(btree)]
    pub org_id: String,
    pub name: String,
    #[index(btree)]
    pub app_id: String,
    pub is_default: bool,
    pub is_protected: bool,
    pub color: String,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
